### DISCUSSION TITLE
Custom Script - Create Data Variables Dynamically

### DISCUSSION THREAD

**Discussion by ragreyling** [netbox-community/netbox#8625] [Q&A]
Is it all possible to create data variables dynamically based on user form input?

For e.g. let's extend on the example provided on the Wiki where the script asks for the number of switches to create. What if you want to create 2 switches with each switch being a different manufacturer and model? For example a 12 port and a 24 port?

If two switches are indicated, this will display 4 form fields: 2 manufacturer fields and 2 model fields, i.e. one manufacturer field and one model field for each switch.

If something like this isn't possible, any suggestions on a workaround for a scenario like this with the current implementation of Custom Script?

Thank you in advance!

**Comment by jeremystretch** [comment]
You can always pre-populate a fixed number of fields, or write some logic to support pattern expansion. However, as soon as you find yourself in the pattern of detailing multiple objects, the built-in CSV import functionality is probably more appropriate.

**Comment by davidgwatkins** [comment]
I know this one is old, but I found it during a Google.  What if I have a custom script to add a device, and from the same script, I want to prompt the user to add a varying number of host devices by selecting the available interfaces in the device series.  Would need some way to read from one selection and use that input to drive additional subsequent selections (either prompting for host device definitions for each interface from the selected device, or adding two drop downs to select the host and the associated available interfaces).  Any thoughts?
